src/main/java/org/opensearch/ad/indices/AnomalyDetectionIndices.java

```diff
@@ -304,13 +305,57 @@ public static String getCheckpointMappings() throws IOException {
         return Resources.toString(url, Charsets.UTF_8);
     }
 
+    /**
+     * Determine if index exists
+     *
+     * @param indexName the name of the index
+     * @return true if index exists
+     */
+    public boolean indexExists(String indexName) {
+        GetIndexRequest getindexRequest = new GetIndexRequest(indexName);
+
+        CompletableFuture<Boolean> existsFuture = new CompletableFuture<>();
+        sdkRestClient.indices().exists(getindexRequest, ActionListener.wrap(response -> { existsFuture.complete(response); }, exception -> {
+            existsFuture.completeExceptionally(exception);
+        }));
+
+        Boolean existsResponse = existsFuture
+            .orTimeout(AnomalyDetectorSettings.REQUEST_TIMEOUT.get(environmentSettings).getMillis(), TimeUnit.MILLISECONDS)
+            .join();
```

Comment:
Do we need any exception handling here, even if it's just catching and rethrowing with some more information?  OK if the answer is no, just wondering what the user experience will be.

(Repeat comment for most of these calls.)

Reply:
I dont think its necessary as the only way this request fails is by timing out, since even if the index/alias does not exist, it would follow the happy path (for lack of a better word). There's no other information to provide.